dask_cuda/initialize.py

```diff
@@ -25,10 +25,13 @@ def initialize(
     starting up Dask workers which create a variety of threads.
 
     To ensure UCX works correctly, it is important to ensure it is initialized with
-    the correct options. This is important for scheduler, workers, and client. This
-    function will ensure that based on the flags and options passed by the user.
+    the correct options. This is especially important for the scheduler and client,
+    which cannot be configured to use UCX with arguments like ``LocalCUDACluster``
+    and ``dask-cuda-worker``. This function will ensure that they are provided a UCX
+    configuration based on the flags and options passed by the user.
```

Comment:
But this function can't be used with `dask-scheduler`, so I don't think it really applies to that, only to the client. The scheduler will still have to rely on environment variables, so I would remove it from here.

Reply:
If the scheduler still relies upon environment variables, does that mean that they must be set explicitly when running Python code?

I had initially removed the `os.environ[...]` calls from the `LocalCUDACluster` example as I thought all non-cluster/worker UCX config was handled through `initialize(...)` - I can add those back in if that's not the case.

Reply:
We have 4 different types:

1. `LocalCUDACluster`: internally sets everything correctly to scheduler and workers, and also does that if `Client` is in the same process being instantiated after the cluster;
2. `dask-scheduler` (CLI): must set UCX options via environment variables (or `distributed.yaml`);
3. `dask-cuda-worker` (CLI): must  set UCX options via command-line arguments;
4. `Client` (standalone process): must set UCX options via environment variables (I think `distributed.yaml` also works here) or may use `dask_cuda.initialize`.

So what I meant was that for case 2 we can't use `dask_cuda.initialize` because it's a standalone process started via CLI. On the other hand, for `Client` we can either choose to use `dask_cuda.initialize` or environment variables, thus I think it only makes sense documenting for use with the client, so I would change the sentence above to:

"This is especially important for the ~scheduler and~ client, ..."